store/tikv/gcworker/gc_worker.go

```diff
@@ -561,14 +593,11 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 		if gcResp.GetError() != nil {
 			return errors.Errorf("unexpected gc error: %s", gcResp.GetError())
 		}
-		regions++
 		key = loc.EndKey
-		if len(key) == 0 {
+		if len(key) == 0 || bytes.Compare(key, endKey) >= 0 {
```

Comment:
I don't think L596-L599 should exist.

Reply:
I want the func `doGCForOneRegion` used to cleanup one region, if we remove this condition, the func is confused.